### PR TITLE
[13.0][ADD] purchase_order_qty_change_no_recompute: Prevent recompute if only quantity has changed in purchase order line

### PR DESCRIPTION
**Pull request by victoralmau**
Prevent recompute if only quantity has changed in purchase order line.

Locked by:
- [x] `purchase_line_procurement_group` https://github.com/OCA/purchase-workflow/pull/1257

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT31537

### CODE REVIEW

**Comment by victoralmau** [comment]
The current travis error is from the `purchase_line_procurement_group` addon and is not related to this PR.

**Review by chienandalu — COMMENTED** [review]
What if the onchange is legit? Shouldn't we flag this exceptional behavior only when the price is changed on hand?

File: purchase_order_qty_change_no_recompute/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Purchase Order Qty change no recompute",
+    "summary": "Prevent recompute if only quantity has changed in purchase order line",
+    "version": "13.0.1.0.1",
```
Comment (joao-p-marques) [inline purchase_order_qty_change_no_recompute/__manifest__.py]:
```suggestion
    "version": "13.0.1.0.0",
```

File: purchase_order_qty_change_no_recompute/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+This module prevent recompute if only quantity has changed in purchase order line.
```
Comment (joao-p-marques) [inline purchase_order_qty_change_no_recompute/readme/DESCRIPTION.rst]:
```suggestion
This module prevents recompute if only quantity has changed in purchase order line.
```

**Review by joao-p-marques — COMMENTED** [review]
As @chienandalu said, maybe ignoring the onchange completely is a bit dangerous, as it may introduce some other changes that make sense, like the `date_planned` (https://github.com/odoo/odoo/blob/13.0/addons/purchase/models/purchase.py#L715).

So what if, instead of forcing the onchange to be skipped, you overwrote it and applied only what makes sense here?

Apart from that, just some minor comments to the README and manifest.

File: purchase_order_qty_change_no_recompute/readme/DESCRIPTION.rst (hunk 2 above)
Comment (pedrobaeza) [reply purchase_order_qty_change_no_recompute/readme/DESCRIPTION.rst]:
```suggestion
This module prevents to recompute if only quantity has changed in purchase order line.
```

**Comment by pedrobaeza** [comment]
The approach to fade out completely the onchange is on purpose, as we don't want any data to be changed when changing quantity, so it's OK to leave it as is.

**Review by chienandalu — APPROVED** [review]
Ok, maybe a safeguard to skip this could be added with a context. Otherwise :+1:

**Comment by pedrobaeza** [comment]
Travis is red. Please check.

**Comment by victoralmau** [comment]
Sorry, it is related to the `purchase_line_procurement_group` addon https://github.com/OCA/purchase-workflow/pull/1234#issuecomment-909980018

File: purchase_order_qty_change_no_recompute/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    def _onchange_eval(self, field_name, onchange, result):
+        """Remove the trigger for the undesired onchange method with this field.
+        We have to act at this place, as `_onchange_methods` is defined as a
+        property, and thus it can't be inherited due to the conflict of
+        inheritance between Python and Odoo ORM, so we can consider this as a HACK.
+        """
+        if field_name == "product_qty":
```
Comment (pedrobaeza) [inline purchase_order_qty_change_no_recompute/models/purchase_order.py]: Please add a safeguard for tests evaluating `config["test_enable"]` and only acting if a special context key is passed.

**Review by joao-p-marques — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

File: purchase_order_qty_change_no_recompute/models/purchase_order.py (hunk 3 above)
Comment (victoralmau) [reply purchase_order_qty_change_no_recompute/models/purchase_order.py]: Do you think we should apply this change also to the `sale_order_qty_change_no_recompute` addon?

Comment (pedrobaeza) [reply purchase_order_qty_change_no_recompute/models/purchase_order.py]: Yeah, that would prevent the problem you have with the other module in tests.

**Comment by joao-p-marques** [comment]
@victoralmau Travis is still red. Can you please check it?

Looks like a rebase could solve it...

**Comment by victoralmau** [comment]
The travis bug is related to the `purchase_line_procurement_group` addon and is not resolved yet.

In another PR https://github.com/OCA/purchase-workflow/pull/1234 I have added a commit that IMO solves the problem, so if we merge that PR first, then you could rebase and merge this one as well.

**Comment by pedrobaeza** [comment]
Please rebase

**Comment by victoralmau** [comment]
Rebase done, ready to merge now.

**Pull request merged**